src/main/java/sparksoniq/spark/iterator/flowr/OrderByClauseSparkIterator.java

```diff
@@ -242,6 +255,18 @@ private void setAllLocalResults() {
                         } else {
                             // do nothing, type is already set to integer
                         }
+                    } else if (currentColumnType.equals("duration")) {
```

Comment:
I am not sure I understand what is going on here.

If columnType is not a duration, for example, will the unexpected type exception in the last else if no longer be thrown?

Also, the puts are idempotents (except yearMonthDuration replaced with dayTimeDuration, which may be an overlook) and seem to write back the value that was already there.

What issue is being addressed here that requires these extra else ifs?

Reply:
You are right, thank you for pointing this out to me. 
Indeed all those else-ifs are not necessary. Although, for durations, I have to introduce something similar to what is being done for numbers. 
Now the UnexpectedTypeException is thrown if there are sorting keys of different types (see _test_files/runtime-spark/DataFrames/OrderbyClauseError.jq_)